### DISCUSSION TITLE
Solution to undropable uniques

### DISCUSSION THREAD

**Discussion by ghost** [diasurgical/DevilutionX#5841] [Ideas]
Alright, so i have an idea on how to solve this issue. It ain't pretty, but it will probably work.

![Undropable uniques](https://user-images.githubusercontent.com/53155671/221582154-ffe1b5c6-95ae-4794-a899-1c56bae3a7b1.png)
So the problem these items have is that two or more exist with the same base item type and qlvl.

**Solution?**
Make a new set of base items that is identical in terms of statistics. This means that there will be two base item maces for example. To compensate for having twice as many of effectively the same item, the drop chances should probably be halved for both the original and the copy-base item.

Downside? Some items will be much harder to find - This includes the beautiful Civerb's cudgel. Also, it sounds like a lot of work and hassle.

**Comment by DakkJaniels** [comment]
You forgot the downside of item morphing

**Reply by ghost** [reply]
Doesn't that only happen if they share the same base item?

I'm talking about making an additional set of base items with identical statistics. I'll use maces as an example.

We have base item Mace 1 and base item Mace 2.

We keep Civerb's cudgel in Mace 1 and put Crackrust in Mace 2. Now they no longer have the same base item as far as the game is concerned, although their base statistics are unchanged.

**Reply by AJenbo** [reply]
If you add new items to the system it will likely alter the algoritme which causes all items in the game to morph. This is why multiplayer, singleplayer and Hellfire items are not interchangeable.

**Reply by ghost** [reply]
Ah, i see. Then my suggestion will be very difficult to implement.

**Reply by DakkJaniels** [reply]
It's not difficult - if you don't care about causing players' existing items morphing or backwards compatibility, then it's fine, but there are better ways to do this than making new base items, as @qndel linked already.

**Reply by ghost** [reply]
> It's not difficult

Well, since i care about item morphing it will be ;)

I didn't know about the other ways. I was not sure why they are still unfindable in DevX.

**Reply by DakkJaniels** [reply]
Well it's not difficult then, it's impossible lol

**Reply by AJenbo** [reply]
It's possible that setting them to never drop might allow them to not affect the existing item algoritme, but I haven't tested. You would then need some trick to swap the base item if one of the problematic unique items drop.

**Comment by qndel** [comment]
how about https://github.com/diasurgical/devilutionX/pull/2930
